cluster/kube-push.sh

```diff
@@ -27,10 +27,61 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/cluster/kube-env.sh"
 source "${KUBE_ROOT}/cluster/${KUBERNETES_PROVIDER}/util.sh"
 
-echo "Updating cluster using provider: $KUBERNETES_PROVIDER"
+function usage() {
+  echo "${0} [-m|-n <node id>] <version>"
+  echo "  Updates Kurnetes binaries. Can be done for all components (by default), master(-m) or specified node(-n)."
+  echo "  If the version is not specified will try to use local binaries."
+  echo "  Warning: upgrading single node is experimental"
+}
+
+push_to_master=false
+push_to_node=false
+
+while getopts "mn:h" opt; do
+  case ${opt} in
+    m)
+      push_to_master=true;;
+    n)
+      push_to_node=true
+      node_id="$OPTARG";;
+    h)
+      usage
+      exit 0;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit 1;;
+  esac
+done
+shift $((OPTIND-1))
+
+if [[ "${push_to_master}" == "true" ]] && [[ "${push_to_node}" == "true" ]]; then
+  echo "Only one of options -m -n should be specified"
```

Comment:
Yet for other platform, it could be good enough to run master components w/o kubelet (bare-metal ubuntu for instance). Hence my guess is that *master* is not a strict term technologically.